storage/remote/metadata_watcher.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/scrape"
+)
+
+// MetadataAppender is an interface used by the Metadata Watcher to send metadata read from the scrape manager on to somewhere else.
+type MetadataAppender interface {
+	AppendMetadata(context.Context, []scrape.MetricMetadata)
```

Comment:
The problem here is that both `Append` and `AppendMetadata` are on the `QueueManager`. To be fair, I don't like the verb append here as we're not appending metadata like we do for samples but at least is consistent.

Reply:
> we're not appending metadata like we do for samples but at least is consistent

The idea is that we eventually will be though. FWIW I think we should keep the current name, since if we don't want to block `QueueManager` or anyone using RW as a library from implementing both interfaces via a single struct.